### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by Nekotekina**
1) Some headers simplified for better compilation time
2) Some templates simplified for smaller executable size
3) Eliminate std::future to fix compilation for mingw64
4) PKG installation can be cancelled now
5) cellGame fixes

### CODE REVIEW

**Comment by danilaml** [comment]
@Nekotekina are these fixes needed for mingw-clang only? It seems to be the fault on their side.

File: rpcs3/Emu/Cell/Modules/cellGcmSys.cpp [hunk 1]
```diff
@@ -516,15 +516,15 @@ s32 cellGcmSetPrepareFlip(PPUThread& ppu, vm::ptr<CellGcmContextData> ctxt, u32
 
 	if (ctxt.addr() == gcm_info.context_addr)
 	{
-		vm::_ref<CellGcmControl>(gcm_info.control_addr).put += 2 * sizeof(u32);
+		vm::_ref<CellGcmControl>(gcm_info.control_addr).put += 2 * 4;
 	}
 #else
 	// internal compiler error, try to avoid it for now
 	u32 command_size = rsx::make_command(ctxt->current, GCM_FLIP_COMMAND, id);
```
Comment (tambry) [inline rpcs3/Emu/Cell/Modules/cellGcmSys.cpp]: Has this been fixed yet?

**Comment by SakataGintokiYT** [comment]
##### CellGame :
{PPU[0x10] Thread (main_thread) [0x00742ee8]} class fmt::exception thrown: cellDiscGameRegisterDiscChangeCallback(): 
(in file C:\rpcs3\rpcs3\Emu\Cell\Modules\cellGame.cpp:688)

{PPU[0x2d] Thread (mainThread) [0x009b66f8]} class fmt::exception thrown: cellDiscGameGetBootDiscInfo(): 
(in file C:\rpcs3\rpcs3\Emu\Cell\Modules\cellGame.cpp:683)

{PPU[0x11] Thread (main_thread) [0x01865700]} class fmt::exception thrown: cellGameDataGetSizeKB(): 
(in file C:\rpcs3\rpcs3\Emu\Cell\Modules\cellGame.cpp:138)

PPUInterpreter :
{PPU[0x10] Thread (main_thread) [0x000f277c]} class std::runtime_error thrown: Trap!
(in file C:\rpcs3\rpcs3\Emu\Cell\PPUInterpreter.cpp:1759)

{PPU[0x11] Thread (main_thread) [0x000ff2e4]} class std::runtime_error thrown: MFVSCR
(in file C:\rpcs3\rpcs3\Emu\Cell\PPUInterpreter.cpp:93)
________________________________________________________________________________

##### Utilities\File.cpp :
{PPU[0x9f] Thread (CRI FS File Access 1) [0x010da194]} class fmt::exception thrown: Win32 error: 998.
(in file C:\rpcs3\Utilities\File.cpp:793)

{PPU[0x10] Thread (main_thread) [0x004cee28]} class fmt::exception thrown: Unknown Win32 error: 267.
(in file C:\rpcs3\Utilities\File.cpp:1093)

{PPU[0x11] Thread (main_thread) [0x008de17c]} class fmt::exception thrown: Unknown Win32 error: 123 (E:/Emulatory I Gry/PS3/Rpcs3-2016;04;18-x64/dev_hdd0/game/NPJA90185/USRDIR/e:/dev_hdd0/game/NPJA90185/USRDIR/Data/Main.pssg).
(in file C:\rpcs3\Utilities\File.cpp:362)

{PPU[0x74] Thread (InitThread) [0x0068ad30]} class fmt::exception thrown: Unknown Win32 error: 123 (E:/Emulatory I Gry/PS3/Rpcs3-2016;04;18-x64/dev_hdd0/game/NPEB00621/USRDIR/e:///dev_hdd0/game/NPEB00621/USRDIR/CloudMedia/fonts/fry.otf).
(in file C:\rpcs3\Utilities\File.cpp:362)

{PPU[0x11] Thread (main_thread) [0x0030be20]} class fmt::exception thrown: Unknown Win32 error: 123 (E:/Emulatory I Gry/PS3/Rpcs3-2016;04;18-x64/dev_hdd0/game/NPJB00066/USRDIR/e:/dev_hdd0/game/NPJB00066/USRDIR/system/PostFilter.PSSG.gz).
(in file C:\rpcs3\Utilities\File.cpp:362)

________________________________________________________________________________

##### cellGcmSys:
{PPU[0xd0] Thread (RenderThread) [0x00c0a9b4]} class fmt::exception thrown: cellGcmSetDefaultCommandBufferAndSegmentWordSize(): 
(in file C:\rpcs3\rpcs3\Emu\Cell\Modules\cellGcmSys.cpp:1143)

{PPU[0x11] Thread (main_thread) [0x005d4dfc]} class fmt::exception thrown: _cellGcmFunc2(): 
(in file C:\rpcs3\rpcs3\Emu\Cell\Modules\cellGcmSys.cpp:1212)

{PPU[0x11] Thread (main_thread) [0x00173af0]} class fmt::exception thrown: _cellGcmFunc2(): 
(in file C:\rpcs3\rpcs3\Emu\Cell\Modules\cellGcmSys.cpp:1212)

________________________________________________________________________________

##### Memory\vm.cpp:
{PPU[0x10] Thread (main_thread) [0x01446234]} class fmt::exception thrown: vm::stack_push(): Stack overflow (size=0x8, align=0x4, SP=0x7ff70187bb20, stack=*0xd0000000)
(in file C:\rpcs3\rpcs3\Emu\Memory\vm.cpp:885)

{PPU[0x10] Thread (main_thread) [0x0001bb98]} class fmt::exception thrown: vm::block_t::try_alloc::<lambda_69e016b27ca6d4be3b13e173fd48d11f>::operator ()(): Unexpected memory amount used (0xfb100000)
(in file C:\rpcs3\rpcs3\Emu\Memory\vm.cpp:613)

{PPU[0x11] Thread (main_thread) [0x0006498c]} class fmt::exception thrown: vm::block_t::try_alloc::<lambda_69e016b27ca6d4be3b13e173fd48d11f>::operator ()(): Unexpected memory amount used (0xfd600000)
(in file C:\rpcs3\rpcs3\Emu\Memory\vm.cpp:613)

{rsx::thread} class fmt::exception thrown: vm::page_protect(): Invalid arguments (addr=0x21120000, size=0x0)
(in file C:\rpcs3\rpcs3\Emu\Memory\vm.cpp:436)

________________________________________________________________________________

##### SPUThread.cpp:
{SPU[0x33] Thread (CellSpursKernel2) [0x1ff00]} class fmt::exception thrown: SPUThread::stop_and_signal(): Unknown STOP code: 0x0 (Out_MBox is empty)
(in file C:\rpcs3\rpcs3\Emu\Cell\SPUThread.cpp:1344)

________________________________________________________________________________

##### PPUModule.cpp:
{PPU[0xc0] Thread (_libsail-control) [0x0099ac70]} class fmt::exception thrown: Function not registered (index 3011)
(in file C:\rpcs3\rpcs3\Emu\Cell\PPUModule.cpp:57)

**Comment by raven02** [comment]
Just tested with interpreter (fast) , the speed seems to be dropped to half .

**Comment by Nekotekina** [comment]
What did you test? This PR wasn't supposed to affect execution speed anyway.

**Comment by raven02** [comment]
I tested with Dragon Ball Z Burst limit .

**Comment by paulsapps** [comment]
Unless std::future changes have some how made multi threaded performance lower I can't see what would have caused this..

**Comment by raven02** [comment]
I try to exlcude std::future in the RSXThread one but seems not helping .Screenshot here before this commit and after.

- Optimizations #1680


- GSL moved from stdafx.h (#1676)

**Comment by danilaml** [comment]
@raven02 can you try using vs15 profiler to see what might've changed?

**Comment by raven02** [comment]
This is the CPU profiler in VS2015

**Comment by danilaml** [comment]
@raven02 is it for this PR of before? Comparing them might be interesting.

**Comment by paulsapps** [comment]
I think you needed to expand out that high usage function to break it down more

**Pull request merged**